substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/jfr/JfrTypeRepository.java

```diff
@@ -102,15 +102,20 @@ private TypeInfo collectTypeInfo(boolean flushpoint) {
         return typeInfo;
     }
 
+    boolean isClassGenerated(Class<?> clazz) {
+        return clazz != null && clazz.getPackage() != null && clazz.getPackage().getName().equals("jdk.internal.reflect") && clazz.getName().contains("GeneratedSerializationConstructorAccessor");
```

Comment:
Previously, I needed to determine which classes were generated in order  to adjust the assertions. This was because there were cases where generated classes have the same package as non-generated classes, however their module is different (the unnamed module).

This means there was the possibility for  a single package name (which was being used as the key for deduplication) to correspond to multiple modules.  

In hotspot, no special handling is done for generated classes. Instead, each class has a pointer back to its package instance which then corresponds to a single module. This means there is a one-one relationship instead of one-many. 

To replicate the same behaviour here, we need a unique key which corresponds to exactly one package-module pair.  To do this I've decided to use the concatenation of the concatenate the package and module names.